### PR TITLE
Common static server address : Fix #424

### LINKED ISSUE #424: [Common] Add the possibility to have a static server address
### Prerequisites

- [ ] I have written a descriptive issue title
- [ ] I have verified that I am running the latest version of the COMET-WEB
- [ ] I have searched [open](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

- Create a service, that reads static server address from config file to connect to the app

### Logs

  - [ ] There are no logs regarding the issue
  - [ ] I have attached a log file or included the relevant section in this issue  
  
### System Configuration

- COMET version:
  - [ ] COMET-WEB: 
  - [ ] Other: 
- Web Services:
  - [ ] COMET Web Services version:

### PR DESCRIPTION
**Pull request by nabilatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fix #424 

Add the possibility to connect to the server using configuration file.

### CODE REVIEW

File: COMET.Web.Common/Extensions/WebAssemblyHostExtensions.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@
 namespace COMET.Web.Common.Extensions
 {
     using COMET.Web.Common.Services.ConfigurationService;
-
+    using COMET.Web.Common.Services.ServerConnexionService;
     using Microsoft.AspNetCore.Components.WebAssembly.Hosting;
```
Comment (samatrhea) [inline COMET.Web.Common/Extensions/WebAssemblyHostExtensions.cs:30]: please add newline between COMET and Microsoft namespaces

**Review by samatrhea — APPROVED** [review]

File: COMET.Web.Common.Tests/Components/IndexComponentTestFixture.cs [hunk 2]
```diff
@@ -40,7 +40,8 @@ namespace COMET.Web.Common.Tests.Components
     using COMET.Web.Common.Model;
     using COMET.Web.Common.Services.ConfigurationService;
     using COMET.Web.Common.Services.RegistrationService;
-    using COMET.Web.Common.Services.SessionManagement;
+	using COMET.Web.Common.Services.ServerConnectionService;
```
Comment (antoineatrhea) [inline COMET.Web.Common.Tests/Components/IndexComponentTestFixture.cs:43]: indentation issue in the file

File: COMET.Web.Common.Tests/Components/LoginTestFixture.cs [hunk 3]
```diff
@@ -30,7 +30,8 @@ namespace COMET.Web.Common.Tests.Components
     using COMET.Web.Common.Components;
     using COMET.Web.Common.Enumerations;
     using COMET.Web.Common.Model.DTO;
-    using COMET.Web.Common.Services.SessionManagement;
+	using COMET.Web.Common.Services.ServerConnectionService;
```
Comment (antoineatrhea) [inline COMET.Web.Common.Tests/Components/LoginTestFixture.cs:33]: indentation issue in the file

File: COMET.Web.Common.Tests/Components/SingleIterationApplicationTemplateTestFixture.cs [hunk 4]
```diff
@@ -37,7 +37,8 @@ namespace COMET.Web.Common.Tests.Components
     using COMET.Web.Common.Components.Selectors;
     using COMET.Web.Common.Extensions;
     using COMET.Web.Common.Services.ConfigurationService;
-    using COMET.Web.Common.Services.SessionManagement;
+	using COMET.Web.Common.Services.ServerConnectionService;
```
Comment (antoineatrhea) [inline COMET.Web.Common.Tests/Components/SingleIterationApplicationTemplateTestFixture.cs:40]: indentation issue in the file

File: COMET.Web.Common.Tests/Services/ServerConnectionService/ServerConnectionServiceTestFixture.cs [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+// 
+//  </copyright>
+//  --------------------------------------------------------------------------------------------------------------------
+namespace COMET.Web.Common.Tests.Services.ServerConnectionService
+{
+	using System.Net;
```
Comment (antoineatrhea) [inline COMET.Web.Common.Tests/Services/ServerConnectionService/ServerConnectionServiceTestFixture.cs:27]: indentation issue in the file

File: COMET.Web.Common/Components/Login.razor [hunk 6]
```diff
@@ -25,14 +25,17 @@
...
 	<DxFormLayout CaptionPosition="CaptionPosition.Vertical">
-		<DxFormLayoutItem Caption="Source Address:" ColSpanLg="12">
-			<Template>
-				<DxTextBox Id="sourceaddress" @bind-Text="@(this.ViewModel.AuthenticationDto.SourceAddress)"
-				           ClearButtonDisplayMode="DataEditorClearButtonDisplayMode.Auto"
-				           NullText="Enter the url for the server"
-				           Enabled="@string.IsNullOrEmpty(this.RequestedServer)"/>
-			</Template>
-		</DxFormLayoutItem>
+		@if (this.ViewModel.serverConnectionService.ServerAddress == null)
```
Comment (antoineatrhea) [inline COMET.Web.Common/Components/Login.razor:28]: can you use the 'string.IsNullOrEmpty' method instead please?

File: COMET.Web.Common/Components/SingleIterationApplicationTemplate.razor.cs [hunk 7]
```diff
@@ -136,7 +143,7 @@ internal void SetCorrectUrl()
             {
                 currentOptions[QueryKeys.IterationKey] = this.ViewModel.SelectedIteration.Iid.ToShortGuid();
                 currentOptions[QueryKeys.ModelKey] = this.ViewModel.SelectedIteration.IterationSetup.Container.Iid.ToShortGuid();
-                currentOptions[QueryKeys.ServerKey] = this.ViewModel.SessionService.Session.DataSourceUri;
+				currentOptions[QueryKeys.ServerKey] = this.ViewModel.SessionService.Session.DataSourceUri;
```
Comment (antoineatrhea) [inline COMET.Web.Common/Components/SingleIterationApplicationTemplate.razor.cs:146]: indentation issue in the file. can you also remove the option from the url if the static address is defined?

File: COMET.Web.Common/Extensions/WebAssemblyHostBuilderExtensions.cs [hunk 8]
```diff
@@ -29,7 +29,8 @@ namespace COMET.Web.Common.Extensions
     using COMET.Web.Common.Services.ConfigurationService;
     using COMET.Web.Common.Services.NotificationService;
     using COMET.Web.Common.Services.RegistrationService;
-    using COMET.Web.Common.Services.SessionManagement;
+	using COMET.Web.Common.Services.ServerConnectionService;
```
Comment (antoineatrhea) [inline COMET.Web.Common/Extensions/WebAssemblyHostBuilderExtensions.cs:32]: indentation issue in the file

File: COMET.Web.Common/Services/ServerConnectionService/ServerConnectionService.cs [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+// 
+//  </copyright>
+//  --------------------------------------------------------------------------------------------------------------------
+namespace COMET.Web.Common.Services.ServerConnectionService
+{
+    using System;
+    using System.Text.Json;
+
+    using COMET.Web.Common.Model;
+	using COMET.Web.Common.Utilities;
```
Comment (antoineatrhea) [inline COMET.Web.Common/Services/ServerConnectionService/ServerConnectionService.cs:31]: indentation issue in the file

File: COMET.Web.Common/Services/ServerConnectionService/ServerConnectionService.cs [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+            {
+                return;
+            }
+
+            try
+            {
+                var path = ContentPathBuilder.BuildPath(this.ServerConfigurationFile);
+                var jsonContent = await this.http.GetStreamAsync(path);
+                configurations = JsonSerializer.Deserialize<Dictionary<string, string>>(jsonContent);
+                ServerAddress = configurations["ServerAddress"];
```
Comment (antoineatrhea) [inline COMET.Web.Common/Services/ServerConnectionService/ServerConnectionService.cs:88]: missing this qualifier

File: COMET.Web.Common/ViewModels/Components/ILoginViewModel.cs [hunk 11]
```diff
@@ -27,16 +27,22 @@ namespace COMET.Web.Common.ViewModels.Components
 {
     using COMET.Web.Common.Enumerations;
     using COMET.Web.Common.Model.DTO;
+	using COMET.Web.Common.Services.ServerConnectionService;
 
-    /// <summary>
-    /// View Model that enables the user to login against a COMET Server
-    /// </summary>
-    public interface ILoginViewModel
+	/// <summary>
```
Comment (antoineatrhea) [inline COMET.Web.Common/ViewModels/Components/ILoginViewModel.cs:32]: indentation issue in the file

File: COMET.Web.Common/wwwroot/server_configuration.json [hunk 12]
```diff
@@ -0,0 +1,3 @@
+{
+  "ServerAddress": "https://comet.innovetusa.com/"
```
Comment (antoineatrhea) [inline COMET.Web.Common/wwwroot/server_configuration.json:2]: remove the value

File: COMETwebapp.Tests/Pages/ModelDashboard/ModelDashboardTestFixture.cs [hunk 13]
```diff
@@ -35,8 +35,9 @@ namespace COMETwebapp.Tests.Pages.ModelDashboard
     using COMET.Web.Common.Components.Selectors;
     using COMET.Web.Common.Extensions;
     using COMET.Web.Common.Services.ConfigurationService;
-    using COMET.Web.Common.Services.SessionManagement;
-    using COMET.Web.Common.Test.Helpers;
+	using COMET.Web.Common.Services.ServerConnectionService;
```
Comment (antoineatrhea) [inline COMETwebapp.Tests/Pages/ModelDashboard/ModelDashboardTestFixture.cs:38]: indentation issue in the file

File: COMETwebapp.Tests/Pages/ParameterEditor/ParameterEditorTestFixture.cs [hunk 14]
```diff
@@ -36,8 +36,9 @@ namespace COMETwebapp.Tests.Pages.ParameterEditor
     using COMET.Web.Common.Extensions;
     using COMET.Web.Common.Services.ConfigurationService;
     using COMET.Web.Common.Services.NotificationService;
-    using COMET.Web.Common.Services.SessionManagement;
-    using COMET.Web.Common.Test.Helpers;
+	using COMET.Web.Common.Services.ServerConnectionService;
```
Comment (antoineatrhea) [inline COMETwebapp.Tests/Pages/ParameterEditor/ParameterEditorTestFixture.cs:39]: indentation issue in the file

File: COMETwebapp.Tests/Pages/Viewer/ViewerTestFixture.cs [hunk 15]
```diff
@@ -35,8 +35,9 @@ namespace COMETwebapp.Tests.Pages.Viewer
     using COMET.Web.Common.Components.Selectors;
     using COMET.Web.Common.Extensions;
     using COMET.Web.Common.Services.ConfigurationService;
-    using COMET.Web.Common.Services.SessionManagement;
-    using COMET.Web.Common.Test.Helpers;
+	using COMET.Web.Common.Services.ServerConnectionService;
```
Comment (antoineatrhea) [inline COMETwebapp.Tests/Pages/Viewer/ViewerTestFixture.cs:38]: indentation issue in the file

File: COMET.Web.Common/Services/ServerConnectionService/ServerConnectionService.cs [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+
+            if (this.isInitialized)
+            {
+                return;
+            }
+
+            try
+            {
+                var path = ContentPathBuilder.BuildPath(this.ServerConfigurationFile);
+                var jsonContent = await this.http.GetStreamAsync(path);
```
Comment (antoineatrhea) [inline COMET.Web.Common/Services/ServerConnectionService/ServerConnectionService.cs:86]: what if the file does not exists? please make a check based on the reply

**Review by antoineatrhea — CHANGES_REQUESTED** [review]
please verify your indentation setting inside VS and ReSharper. Moreover, don't use the server address inside the url if defined inside the configuration file

**Review by samatrhea — APPROVED** [review]

File: COMET.Web.Common/Services/ServerConnectionService/ServerConnectionService.cs [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+// 
+//  </copyright>
+//  --------------------------------------------------------------------------------------------------------------------
+namespace COMET.Web.Common.Services.ServerConnectionService
+{
+    using System;
+	using System.Net;
```
Comment (antoineatrhea) [inline COMET.Web.Common/Services/ServerConnectionService/ServerConnectionService.cs:28]: still indentation problem in the file

File: COMET.Web.Common/ViewModels/Components/LoginViewModel.cs [hunk 18]
```diff
@@ -27,31 +27,37 @@ namespace COMET.Web.Common.ViewModels.Components
 {
     using COMET.Web.Common.Enumerations;
     using COMET.Web.Common.Model.DTO;
-    using COMET.Web.Common.Services.SessionManagement;
+	using COMET.Web.Common.Services.ServerConnectionService;
```
Comment (antoineatrhea) [inline COMET.Web.Common/ViewModels/Components/LoginViewModel.cs:30]: still indentation problem inside the file

File: COMET.Web.Common/Components/SingleIterationApplicationTemplate.razor.cs [hunk 19]
```diff
@@ -143,12 +143,17 @@ internal void SetCorrectUrl()
             {
                 currentOptions[QueryKeys.IterationKey] = this.ViewModel.SelectedIteration.Iid.ToShortGuid();
                 currentOptions[QueryKeys.ModelKey] = this.ViewModel.SelectedIteration.IterationSetup.Container.Iid.ToShortGuid();
-				currentOptions[QueryKeys.ServerKey] = this.ViewModel.SessionService.Session.DataSourceUri;
-                currentOptions[QueryKeys.DomainKey] = this.ViewModel.SessionService.GetDomainOfExpertise(this.ViewModel.SelectedIteration).Iid.ToShortGuid();
+
+                if (string.IsNullOrEmpty(this.ServerConnectionService.ServerAddress))
```
Comment (antoineatrhea) [inline COMET.Web.Common/Components/SingleIterationApplicationTemplate.razor.cs:147]: still indentation problem inside the file

**Review by antoineatrhea — CHANGES_REQUESTED** [review]
still some indentation problem

**Pull request merged**